### ISSUE TITLE
Lacking privileges to access camera service

### ISSUE THREAD

**Issue by RaduPopescu29** [googlesamples/android-Camera2Basic#42]
When using camera in landscape, closing screen and opening it, crashes the app. Tested on Nexus 5 with 6.0.1
FATAL EXCEPTION: main
 Process: com.example.android.camera2basic, PID: 14591
 java.lang.SecurityException: Lacking privileges to access camera service
 at android.hardware.camera2.utils.CameraBinderDecorator.throwOnError(
 CameraBinderDecorator.java:108)
 at android.hardware.camera2.utils.CameraBinderDecorator$CameraBinderDecoratorListener.
onAfterInvocation(CameraBinderDecorator.java:73)
at android.hardware.camera2.utils.Decorator.invoke(Decorator.java:81)
 at java.lang.reflect.Proxy.invoke(Proxy.java:393)
 at $Proxy1.cancelRequest(Unknown Source)
 at android.hardware.camera2.impl.CameraDeviceImpl.stopRepeating(CameraDeviceImpl.java:926)
 at android.hardware.camera2.impl.CameraCaptureSessionImpl.close(CameraCaptureSessionImpl.java:378)
 at android.hardware.camera2.impl.CameraCaptureSessionImpl$2.onDisconnected(CameraCaptureSessionImpl.java:514)
                                                                                      at android.hardware.camera2.impl.CameraDeviceImpl$7.run(CameraDeviceImpl.java:228)
                                                                                      at android.os.Handler.handleCallback(Handler.java:739)
                                                                                      at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                                      at android.os.Looper.loop(Looper.java:148)
                                                                                      at android.app.ActivityThread.main(ActivityThread.java:5417)
                                                                                      at java.lang.reflect.Method.invoke(Native Method)
                                                                                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                                                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)

**Comment by DanielWorld** [comment]
Right. I got the same issue, too!

**Comment by RoundSparrow** [comment]
I was able to reproduce this bug too.  Can someone confirm this fix:

In the OnResume meethod, before cameraOpen, add a closeCamera() call.

**Comment by ganlaw** [comment]
any word on fixes for this issue?

**Comment by cdrapes17** [comment]
i'm having the same problem. 

adding a closeCamera() call before the cameraOpen() in the OnResume doesn't fix it.

**Comment by rnallamo** [comment]
I am also having same issue, this problem in portrait mode was handled with closeCamera() before openCamera() but in landscape when some how it is not working

**Comment by rnallamo** [comment]
In Camera2Basic example closeCamera() in 
mSurfaceTextureListener (onSurfaceTextureDestroyed). It worked for me.

**Comment by p4block** [comment]
Same problem here. None of the fixes fix it.

In my case I've traced it down to

Camera activity -> Another Activity -INTENT-> Another application -> back button twice to return to Camera activity

**Comment by marcaoas** [comment]
I was having the same problem.
make sure that you are calling the startPreviewingCamera() before you call openCamera

**Comment by jf-rce** [comment]
@p4block please see my latest comment in this thread:

https://github.com/google/cameraview/issues/61

TLDR:

After you call CameraCaptureSession.close(), CameraDevice.close(), ImageReader.close(), etc. when you app is pausing, wait until CameraDevice.StateCallback.onClosed() has fired to open your dialog. If you open it while the Camera is still closing, you may encounter this crash on Android 6.0 only. This was my experience when testing with a slew of devices (4.1, 4.4, 5.0, 5.1, 6.0, 7.1).

**Comment by panxu** [comment]
Have you solve this issue?

**Comment by deepikaopenxcell** [comment]
i am getting this error when on resume call and it is random crash. please check this bug

**Comment by rakshitsoni02** [comment]
anyone got solution about this issue ?

**Comment by sanjaysiliconithub** [comment]
closeCamera() in onSurfaceTextureDestroyed() method of SurfaceTextureListener.

**Comment by yudikarma** [comment]
call close camera in onresume before call open camera, call close camera in surfacetexturelistener on destroy is Not work to me. cause my camera activity in landscape mode . but in portrait mode is work perfect. anyone have some idea ? please help @jf-rce @rnallamo

**Comment by codingjeremy** [comment]
This sample has been migrated to a new location where we can accept Pull Requests (check README for more information). 

[As recommended by GitHub][1], we are closing all issues and pull requests now that this older repo will be archived. 

If you still see this issue in the updated repo, please reopen the issue/PR there. Thank you!

[1]: https://help.github.com/en/articles/archiving-repositories
